pennylane/transforms/__init__.py

```diff
@@ -20,38 +20,32 @@
 Transforms
 ----------
 
-Transforms that act on QNodes
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+How to build a custom PennyLane transform?
+------------------------------------------
 
-These transforms accept QNodes, and return new transformed functions
-that compute the desired quantity.
+The user facing way of creating a transform is to use :class:`pennylane.transform` as a function or a decorator.
+The PennyLane contract for defining your own quantum transform is the following: you need to define a quantum transform
+that takes a :class:`qml.tape.QuantumTape` as first argument and returns a sequence of :class:`qml.tape.QuantumTape` and
+a processing function. The resulting batch of circuits is then executed and the results are post-processed by the
+provided processing function. For more details and an example, read the following page:
 
 .. autosummary::
     :toctree: api
 
-    ~transforms.classical_jacobian
-    ~batch_params
-    ~batch_input
-    ~batch_partial
-    ~metric_tensor
-    ~adjoint_metric_tensor
-    ~specs
-    ~transforms.split_non_commuting
+    ~transforms.core.transform
 
-Transforms that act on quantum functions
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Transforms core
```

Comment:
I don't think a user would know what is meant by "core", but idk what to replace this header with. Transforms infra? Building blocks?

Reply:
'Developer functions'?

Reply:
Yes I like it!